remote/endpoint_watcher.go

```diff
@@ -15,12 +15,12 @@ func newEndpointWatcher(address string) actor.Producer {
 
 type endpointWatcher struct {
 	address string
-	watched map[string]*actor.PID //key is the watching PID string, value is the watched PID
+	watched map[string]*PIDSet //key is the watching PID string, value is the watched PID
```

Comment:
Probably you can just use *actor.PIDSet, there is already a class dedicated for the PID sets.

Reply:
Thank u advise.
thanks for your advice.
I have considered using actor.PID, but in this way, once the business layer uses Watcher.Who, it can not be positioned.Because the actor's address based on protobuf unmarshal is new every time, which is inconsistent with the previous cache of Watcher.Who.
In business layer code:
```golang
//...
	case *messages.Connect:
		log.Printf("Client [%v,%p] connected", msg.Sender,msg.Sender)
		clients.Add(msg.Sender)     //msg.Sender is ptr of actor.PID
		msg.Sender.Tell(&messages.Connected{Message: "Welcome!"})
		context.Watch(msg.Sender) 
	case *actor.Terminated:
			clients.Remove(msg.Who)    //msg.Who ptr is should in clients
			log.Printf("remove after :clients[%+v],who[%p],connected size is %v ", clients,msg.Who,clients.Size())
//...
```
We can catch the object creation in **proto_serializer.go**
```golang
func (protoSerializer) Deserialize(typeName string, bytes []byte) (interface{}, error) {
	protoType := proto.MessageType(typeName)
	if protoType == nil {
		return nil, fmt.Errorf("Unknown message type %v", typeName)
	}
	t := protoType.Elem()    

	intPtr := reflect.New(t)       //<----this is new ptr
	instance := intPtr.Interface().(proto.Message)
	proto.Unmarshal(bytes, instance)

	return instance, nil
}
```
So I have to provide a cache that helps the same actor with the same ptr.

Reply:
Hey, actually in actor.PIDSet, it is using string as key.
 https://github.com/AsynkronIT/protoactor-go/blob/6c50ad304f4f87567c31f30487343b51aa96c7f3/actor/pidset.go#L5-L8
So it might be OK just using it = )